File: src/main/java/com/bdev/hogwarts_api/data/model/event_participant/EventParticipantModel.java
package com.bdev.hogwarts_api.data.model.event_participant;

import com.bdev.hogwarts_api.data.dto.events.EventParticipantStatus;
import com.bdev.hogwarts_api.data.dto.student.StudentReferralSource;
import lombok.Data;

import javax.persistence.*;

@Data
@Entity
@Table(name = "HG_EVENT_PARTICIPANT")
public class EventParticipantModel {
    @Id
    @Column(name = "ID")
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "EVENT_ID")
    private Long eventId;
    @Column(name = "NAME")
    private String name;
    @Column(name = "STATUS")
    @Enumerated(EnumType.STRING)
    private EventParticipantStatus status;
    @Column(name = "PHONE")
    private String phone;
    @Column(name = "REFERRAL_SOURCE")
    @Enumerated(EnumType.STRING)
    private StudentReferralSource referralSource;
}

File: src/main/java/com/bdev/hogwarts_api/data/model/event_participant/EventParticipantModel.kt
package com.bdev.hogwarts_api.data.model.event_participant

import com.bdev.hogwarts_api.data.dto.events.EventParticipantStatus

import javax.persistence.*

@Entity
@Table(name = "HG_EVENT_PARTICIPANT")
open class EventParticipantModel {
    @Id
    @Column(name = "ID")
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long? = null

    @Column(name = "EVENT_ID")
    var eventId: Long? = null
    @Column(name = "NAME")
    var name: String? = null
    @Column(name = "STATUS")
    @Enumerated(EnumType.STRING)
    var status: EventParticipantStatus? = null
    @Column(name = "PHONE")
    var phone: String? = null
    @Column(name = "ENLISTED")
    var enlisted: Boolean? = null
}
